File: backend/app/schemas/intelligence.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class IntelSyncStateResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    source: str
    last_attempt_at: datetime | None
    last_success_at: datetime | None
    last_error: str | None
    details: dict


class IntelSyncTriggerResponse(BaseModel):
    status: str

File: frontend/app/types/intelligence.ts
// Mirrors backend/app/schemas/intelligence.py::IntelSyncStateResponse.
// `details` is a free-form per-source counter bag (e.g. {"catalog_size":
// 1665}) -- informational only, never a fixed shape across sources.
export interface IntelSyncState {
  source: string;
  last_attempt_at: string | null;
  last_success_at: string | null;
  last_error: string | null;
  details: Record<string, unknown>;
}
